call as_view in methodresolver

bluebrown opened the pull request:
Signed-off-by: Nico Braun <rainbowstack@gmail.com>

Fixes #1549 

>**Warning** 
>this is a breaking change
> users will be required to change any search method to a get method, merging _get by id_ and _get collection_ into a single handler

The primary reason for this change is to be able to use method decorators as described here: https://flask.palletsprojects.com/en/2.1.x/views/#decorating-views


Changes proposed in this pull request:

 - change the `MethodResolver` to call `as_view` on the classes

bluebrown commented:
I did not really find out how to write a test for this. All tests have passed. I think this kind of functionality is currently not covered. Like there is no app created to see what the methods return. Or perhaps, I overlooked it.

bluebrown commented:
We, should probably also update the documentation on this.

RobbeSneyders reviewed (commented):
Thanks @bluebrown.

- The changes look good to me in general. I left one small comment on the example.
- We should indeed update the docs on this. Could you have a look?
- As mentioned in the issue, it would be good to still offer a resolver with the old behavior, but we can mark it deprecated.
- The example currently doesn't run on main since we're in the middle of a large refactoring (https://github.com/spec-first/connexion/issues/1489), but the resolver itself does seem to work correctly. I submitted a PR which fixes the blocking issue (https://github.com/spec-first/connexion/pull/1553).
- The current resolver tests indeed only check the generated `operation_id`, which is unchanged. I'll have a deeper look at the tests later to see how we can best add a test for the changed behavior.

bluebrown commented on examples/openapi3/methodresolver/api/petsview.py:
```diff
@@ -5,10 +5,17 @@
+    decorators = [example_decorator]
     pets = {}
 
     def post(self):
```
I have updated the example to use the body param in post and put

bluebrown commented:
- added a way to use the old resolver as `DeprecatedMethodViewResolver`
- added a `class_arguments` parameter to the new `MethodViewResolver` to allow dependency injection
- updated the tests to test both resolvers with the existing tests via parameterized fixture
- updated the example app
- updated the docs

RobbeSneyders commented on connexion/resolver.py:
```diff
@@ -192,7 +192,8 @@ def post(self):
 
-    def __init__(self, *args, **kwargs):
+    def __init__(self, *args,  class_arguments=None, **kwargs):
+        self.class_arguments = class_arguments or {}
```
Could we name this a bit more specific eg. `view_arguments`?
It would also be good to document it here as an argument on top of the documentation you already added.

bluebrown replied:
I think class_arguments is more fitting because the arguments for the view function are also called `class_args` and `class_kwargs` that's because these arguments are passed to the class when instantiating it.

bluebrown replied:
I will try to add a comment to the init function. I had actually one already with example usage, but pre-commit hook didnt like it.

RobbeSneyders reviewed (commented):
Thanks for the changes @bluebrown.
I left one comment, but overall changes look great.

I still need to find the time to have a look at how we can add an additional test for the resulting behavior,
